external-testing/README.md

```diff
@@ -0,0 +1,138 @@
+    * HTTP communcation
+    * `git push`
+    * `git fetch`
+    * `fedpkg clone`
+    * storing files (which?)
+* We still have 2 PRs opened to improve testing via requre in packit and p-s, for months at this point.
+* Our integration tests use pytest features and flexmock heavily - the structure is too complex at this point.
+
+
+## VCRpy vs. requre vs. betamax
+
+| A feature | VCRpy | requre | betamax |
+|-----------|-------|--------|---------|
+| Record & replay for python requests | ✓ | ✓ | ✓ |
+| Match on call stack | ✘ | ✓ | ✘ [5]|
+| Match on URL & request type | ✓ | ✘ | ✓ |
+| Hooks to all requests calls automagically | ✓ | ✓ | ✘ [6] |
+| Calling the same URL twice works as expected | ✓ | ✘ | ? |
+| Works with HTTPS | ✓ | ✓ | ? |
+| Record & replay for arbitrary functions | ✘ | ✓ | ✘ |
+| Filtering sensitive data out of recorded responses | ✓ [1] | ✓ [2] | ✓ [7] |
+| Easy to use | ✓ [3] | ✘ [4] | ? |
+| Actively maintained | ✘ | ✓ | ~ |
+
+* [1] https://vcrpy.readthedocs.io/en/latest/advanced.html#filter-sensitive-data-from-the-request
+* [2] https://requre.readthedocs.io/en/latest/usages/postprocessing.html
+* [3] [You can use a decorator or a context manager](https://vcrpy.readthedocs.io/en/latest/usage.html)
+* [4] [Tenths of lines of code](https://github.com/packit-service/packit/blob/master/tests_recording/__init__.py#L16) [which are very hard to understand](https://github.com/packit-service/packit/blob/master/tests_recording/testbase.py#L22)
+* [5] Betamax has [pluggable matching mechanism](https://betamax.readthedocs.io/en/latest/matchers.html)
+* [6] This is by design: https://github.com/betamaxpy/betamax/issues/125#issuecomment-267405773
+* [7] https://betamax.readthedocs.io/en/latest/serializers.html
+
+### Feelings
+
+These are my personal (@TomasTomecek) comments and opinions:
+* VCRpy's API and documentation is focused on the record & replay use case
+  which makes it very pleasant to work with.
+* requre's documentation and API is centered around implementation details
+  which makes it very configurable, but unfortunately extremely hard to work
+  with - even to the point that you need to understand internals.
+* VCRpy in action:
+  * [Basic usage](https://github.com/packit-service/packit/pull/785)
+  * [Calling the same URL twice](https://github.com/packit-service/ogr/pull/373)
+* The fact that requre is matching replay via call stack makes it less flexible
+  and requires data regeneration more often than VCRpy.
+* In my opinion: VCRpy masters its job very well.
+
+
+## So, what's next?
+
+Hunor asked me to do an in-depth analysis for the possible solutions. We need
+a team discussion to pick one.
+
+
+### Solution 1: Keep using requre and improve it
+
+We can keep using requre as we are now. But given that VCRpy has
+* very neat user experience
+* and matches requests based on their data and not the call stack,
+
+it is very compelling to have VCRpy's experience. This means that if we stick
```

Comment:
That would be a good challenge!

I believe something like requre should exist for Python and would be sad if it dies. 
(I am not saying that should be the main benefit of that solution...;-)